core/src/main/scala/org/ensime/core/RichPresentationCompiler.scala

```diff
@@ -219,35 +220,35 @@ trait RichCompilerControl extends CompilerControl with RefactoringControl with C
   def askNotifyWhenReady(): Unit = ask(setNotifyWhenReady)
 
   // WARNING: be really careful when creating BatchSourceFiles. there
-  // are multiple constructers which do weird things, best to be very
+  // are multiple constructors which do weird things, best to be very
   // explicit about what we're doing and only use the primary
   // constructor. Note that scalac appears to have a bug in it whereby
   // it is unable to tell that a VirtualFile (i.e. in-memory) and a
   // non VirtualFile backed BatchSourceFile are actually referring to
   // the same compilation unit. see
   // https://github.com/ensime/ensime-server/issues/1160
   def createSourceFile(file: File): BatchSourceFile =
-    createSourceFile(SourceFileInfo(file, None, None))
+    createSourceFile(SourceFileInfo(file.toPath, None, None))
   def createSourceFile(path: String): BatchSourceFile =
     createSourceFile(new File(path))
   def createSourceFile(file: AbstractFile): BatchSourceFile =
     createSourceFile(file.file)
   def createSourceFile(file: SourceFileInfo): BatchSourceFile = file match {
     case SourceFileInfo(f, None, None) =>
       new BatchSourceFile(
-        new PlainFile(f.getPath),
```

Comment:
we'd still want to create `PlainFile` if the `Path` points to a real file, not a path within an archive.